client/app/components/items-list/hooks/useItemsListExtraActions.js

```diff
@@ -0,0 +1,74 @@
+import { filter, includes, intersection } from "lodash";
+import React, { useState, useMemo, useEffect, useCallback } from "react";
+import Checkbox from "antd/lib/checkbox";
+import { Columns } from "../components/ItemsTable";
+
+export default function useItemsListExtraActions(controller, listColumns, ExtraActionsComponent) {
+  const [actionsState, setActionsState] = useState({ isAvailable: false });
+  const [selectedItems, setSelectedItems] = useState([]);
+
+  // clear selection when page changes
```

Comment:
The idea was that user should work with whit they see. If we leave selected items on other pages - it creates more issues than solves. User cannot see the whole set of selected items (and therefore may make mistakes when doing stuff in bulk); it will be harder to deselect items as now user has to go through all pages. Also, it's a bit harder from technical side: first of all, current page changes not only from a paginator, but when, say, some tags are selected, and generally it's hard to tell which event caused page change. Also, when changing page, all items shown on previous one are released, and if we'll keep references - next time they won't match newly loaded items, and we'll have two copies of the same object on the same page with no way to keep them in sync. At the end, clearing selection gives better control over count of selected items - performing some update on a lot of items may create a big load on backend. Therefore I decided just to clear selection when page changes - it solves all mentioned issues, and actually it's a subject to discuss - is that a bad UX or not.

Reply:
Upd.: One more technical issue - it's hard to tell when ll items are selected, because queries and dashboards lists are paginated so we don't have a full list of items on a frontend.